_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
+}
+```
+### Scenario 8 - Test assert equal JSON - same json node and values as json object
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+
+return
+try{
+	test:assert-equal-json($actual, $expected)
+} catch($err) {
+ $err
+}
+```
+### Scenario 9 - Test assert equal JSON - same json node and values as json document
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+
+let $actualJson := xdmp:to-json($actual)
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+let $expectedJson := xdmp:to-json($expected)
+
+return
+try{
+	test:assert-equal-json($actualJson, $expectedJson)
+} catch($err) {
+ $err
+}
+```
+### Scenario 10 - Test assert equal JSON - Node mismatch as json document and validate expected error message
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+let $_ := map:put($actual,"c",333)
+
+let $actualJson := xdmp:to-json($actual)
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+let $expectedJson := xdmp:to-json($expected)
+let $errorMessageExpected := "Assert Equal Json failed (ASSERT-EQUAL-JSON-FAILED): " || $actualJson || " " || $expectedJson
+return
+try{
+	test:assert-equal-json($actualJson, $expectedJson)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
+```
+### Scenario 11 - Test assert equal JSON - Node value mismatch as json document and validate expected error message
+```
+xquery version "1.0-ml";
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $actual := json:object()
+let $_ := map:put($actual,"a",111)
+let $_ := map:put($actual,"b",222)
+let $_ := map:put($actual,"c",333)
+
+let $actualJson := xdmp:to-json($actual)
+let $expected := json:object()
+let $_ := map:put($expected,"a",111)
+let $_ := map:put($expected,"b",222)
+let $_ := map:put($expected,"c",444)
+let $expectedJson := xdmp:to-json($expected)
+let $errorMessageExpected := "Assert Equal Json failed (ASSERT-EQUAL-JSON-FAILED): " || $actualJson || " " || $expectedJson
+return
+try{
+test:assert-equal-json($actualJson, $expectedJson)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
```

Comment:
No need to try catch.

Add an explanation that this test fails because the Actual JSON property 'c' had a value of 333, when the expected JSON property 'c' had a value of 444.

I'd also consider changing the sample data to make the failure more obvious:

```
let $expected := object-node {
  "property": "expected value"
}

let $actual := object-node {
   "property": "unexpected value"
}
return test:assert-equal-json($expected, $actual)
```